Offseason-Remade/src/main/java/frc/robot/Commands/ChangeLiftHeight.java

```diff
@@ -0,0 +1,81 @@
+package frc.robot.Commands;
+
+import edu.wpi.first.wpilibj.PIDController;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj.command.Command;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import frc.robot.Robot;
+import frc.robot.PIDSources.LiftPIDSource;
+import frc.robot.RobotConstants.Height;
+
+/** this command sets the lift to the desired height */
+public class ChangeLiftHeight extends Command {
+
+  /** the values of the pid */
+  private double kp, kd, ki;
+
+  /** the time the pid controller should wait on the target */
+  private double waitTime;
+
+  /** how large the range of the target is */
+  private double tolerance;
+
+  /** the height the lift needs to get to */
+  private double desiredHeight;
+
+  private double lastTimeOffTarget;
+
+  private PIDController pidController;
+
+  private LiftPIDSource pidSource;
+
+  public ChangeLiftHeight(double kp, double kd, double ki, double waitTime, double tolerance, Height desiredHeight) {
+    this.kp = kp;
+    this.ki = ki;
+    this.kd = kd;
+    this.waitTime = waitTime;
+    this.tolerance = tolerance;
+    this.desiredHeight = desiredHeight.getHeight();
+  }
+
+  /**
+   * this is used to quickly change values using the shuffleboard while finding
+   * pid values
+   */
+  public ChangeLiftHeight(Height desiredHeight) {
+    this(SmartDashboard.getNumber("kp: ", 0), SmartDashboard.getNumber("ki: ", 0), SmartDashboard.getNumber("kd: ", 0),
+        SmartDashboard.getNumber("waitTime: ", 0), SmartDashboard.getNumber("tolerance: ", 0), desiredHeight);
+  }
```

Comment:
SmartDashboard.getNumber ETC.
you have to PUT NUMBERS befor get them

Reply:
can you help me with this

Reply:
YES